src/Internal/Test/E2E/Feedback/Node.purs

```diff
@@ -3,100 +3,71 @@
 
 -- | React to events raised by the browser
+-- |
+-- | Takes a page and a function which provides you with
+-- | a `wait` `Aff` action, wich, when performed, produce you the next `BrowserEvent`
+-- | or throws an error
 subscribeToBrowserEvents
   :: Toppokki.Page
-  -> (BrowserEvent -> Effect Unit)
+  -> (Aff BrowserEvent -> Aff Unit)
   -> Aff Unit
 subscribeToBrowserEvents page cont = do
   logs <- liftEffect $ Ref.new ""
-  let
-    addLogLine line = Ref.modify_ (flip append (line <> "\n")) logs
-  liftEffect $ Toppokki.onConsole
-    ( mkEffectFn1 \cm -> launchAff_ do
+  eventAVar <- AVar.empty
+
+  let addLogLine line = Ref.modify_ (flip append (line <> "\n")) logs
+
+  liftEffect do
+    flip Toppokki.onConsole page $
+      mkEffectFn1 \cm -> launchAff_ do
         Toppokki.consoleMessageText cm >>= liftEffect <<< addLogLine
-    )
-    page
-  makeAff \f -> do
-    liftEffect $ Toppokki.onPageError
-      ( mkEffectFn1
-          ( \err -> do
-              allLogs <- Ref.read logs
-              Console.log allLogs
-              f $ Left err
-          )
-      )
-      page
-    let
-      -- Accepts a number of attempts left.
-      -- An attempt is successful if we get at least one event.
-      process :: Maybe Int -> Aff Unit
-      process attempts = do
-        events <- getBrowserEvents page
-        continue <- and <$> for events \event -> do
-          void $ liftEffect $ try $ cont event
-          case event of
-            Success -> pure false
-            Failure err -> liftEffect $ throw err
-            _ -> pure true
-        if continue then do
-          delay $ Milliseconds $ 1000.0
-          if Array.length events == 0 && attempts /= Just 0 then
-            process (flip sub one <$> attempts)
-          else if attempts == Just 0 then liftEffect $ f $ Left $ error
-            "Timeout reached when trying to connect to CTL Contract running\
-            \ in the browser. Is there a Contract with E2E hooks available\
-            \ at the URL you provided? Did you forget to run `npm run \
-            \e2e-serve`?"
-          else process Nothing
-        else pure unit
 
-    processFiber <- Ref.new Nothing
-    launchAff_ do
-      liftEffect <<< flip Ref.write processFiber <<< Just =<< forkAff do
-        try (process (Just firstTimeConnectionAttempts)) >>= liftEffect <<< f
-    pure $ Canceler \e -> do
-      liftEffect (Ref.read processFiber) >>= traverse_ (killFiber e)
-  where
-  -- How many times to try until we get any event?
-  firstTimeConnectionAttempts :: Int
-  firstTimeConnectionAttempts = 10
+    flip Toppokki.onPageError page $
+      mkEffectFn1 \err -> do
+        allLogs <- Ref.read logs
+        Console.log allLogs
+
+        let errStr = "Page error occured: " <> message err <> "\n" <> allLogs
+        AVarSync.kill (error errStr) eventAVar
+
+  let
+    -- Asyncronously gets browser events and pushes
+    -- to the `eventVAar` one by one.
+    watcher = fix \this -> do
```

Comment:
FWIW You don't need to fix under monadic recursion for `Aff`, ie you can use `watcher` directly and recursively here (and in the other places you have used fix, if you gave them names)

Reply:
Sure. That's the matter of taste. I prefer to use `fix`.

Reply:
This is just less verbose, imho.

foo bar baz quux = something *> foo bar baz quux 
foo bar baz quux = fix \this -> something *> this
foo bar baz quux = let this = foo bar baz quux in something *> this